src/wast-lexer.cc

```diff
@@ -24,7 +24,9 @@
 #include "wabt/lexer-source.h"
 #include "wabt/wast-parser.h"
 
-#define ERROR(...) parser->Error(GetLocation(), __VA_ARGS__)
+#define ERROR(...) \
+  if (parser)      \
+  parser->Error(GetLocation(), __VA_ARGS__)
```

Comment:
But won't this mean that such errors can/will be lost?

Reply:
Yes, now that strings can be part of a `reserved` token, we would lose the log message if an invalid string is found while running NoTrailingReservedChars(). I think the best/cleanest fix for this is https://github.com/WebAssembly/wabt/pull/2013, which just makes the WastLexer work similar to a WastParser (it receives an Errors* on construction and stores it so it can log errors from any context).